pkg/controller/s2irun/s2irun_controller.go

```diff
@@ -24,6 +24,10 @@ import (
 	devopslisters "kubesphere.io/kubesphere/pkg/client/listers/devops/v1alpha1"
 )
 
+/**
+s2irun-controller used to handle s2irun's delete logic.
+s2irun creation and operation provided by s2ioperator
```

Comment:
so we need s2irun-controller and s2ioperator, that's wired. How about move s2ioperator into controller-manager, but also independent s2ioperator so people who need only s2irun can deploy without whole kubesphere package.

Reply:
Currently s2ioperator has a separate e2e test, but ks does not.
Migration also leads to the need to maintain two parts of the code.
There are two parts of the controller because for b2i, the ks API provides the function of uploading artifacts, while s2ioperator itself does not support uploading files, he will only use the link for file download.